### ISSUE TITLE
Dialogs wouldn't activate the MainWindow

### ISSUE THREAD

**Issue by hshahdoost** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#367]
When the window is deactivated, you can't work with popups, even clicking on them wouldn't make the popup active. It can be solved simply by adding **OnPreviewMouseDown** to DialogHost.cs

Something like this would do the job

```
protected override void OnPreviewMouseDown(MouseButtonEventArgs e)
{
     base.OnPreviewMouseDown(e);
     if (Application.Current != null) Application.Current.MainWindow.Activate();
}
```

Please consider that this is a serious problem for full screen applications, since the user would not be aware that the main window is deactive

**Comment by ButchersBoy** [comment]
I'm willing to make this change, with a slight tweak, but I can't reproduce the actual issue in the demo...how exactly are you doing this?

**Comment by cpyfferoen** [comment]
I was having the same behavior - in my MainWindow I had DialogHost Wrapped in a Grid - and that was causing weird full screen behavior. Changed it to match the Demo (not wrapped) and works great now

**Comment by Keboo** [comment]
It appears this issue has been fixed in latest 4.2.1 release
